### PR TITLE
Add Large Tensor Test for linalg_syrk

### PR DESCRIPTION
**Pull request by Zha0q1**
## Description ##
Check if syrk_batch works correctly with large tensors. This test will fail with the current code base; https://github.com/apache/incubator-mxnet/pull/18752 should fix it.

TODO:
1. make test function naming consistent with other large tensor tests (need to rebase after any of those tests are merged)
2. merge this only after the fix has been merged

This test passes on both BLAS int32 and 64 builds.

```
ubuntu@ip-172-31-43-103:~$ MXNET_TEST_COUNT=10000 nosetests --logging-level=DEBUG --verbose -s mxnet/tests/nightly/test_large_array.py:test_linalg_operators
test_large_array.test_linalg_operators ... [23:14:57] ../src/storage/storage.cc:198: Using Pooled (Naive) StorageManager for CPU
ok

----------------------------------------------------------------------
Ran 1 test in 637.245s

OK
```

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by access2rohit** [comment]
@Zha0q1 can you fix this issue with your PR "This branch is out-of-date with the base branch"

**Comment by Zha0q1** [comment]
> @Zha0q1 can you fix this issue with your PR "This branch is out-of-date with the base branch"

Fixed.

File: tests/nightly/test_large_array.py [hunk 1]
```diff
@@ -1791,6 +1792,29 @@ def test_sparse_dot():
...
+        # test both forward and backward
+        # batch syrk will be applied to the last two dimensions
+        A = nd.zeros((2, LARGE_SQ_X, LARGE_SQ_X))
+        for i in range(LARGE_SQ_X):
+            A[0,i,i] = 1
+            A[1,i,i] = 0.1
+        A.attach_grad()
+        with mx.autograd.record():
+            out = nd.linalg.syrk(A, alpha=2, transpose=False)
+        for i in range(LARGE_SQ_X):
```
Comment (access2rohit) [inline tests/nightly/test_large_array.py:1806]: You can check in 2 places in (0,y,y) and (1,y,y). No need to check in 70000 locations

File: tests/nightly/test_large_array.py [hunk 2]
```diff
@@ -1791,6 +1792,29 @@ def test_sparse_dot():
...
+            assert out[0,i,i] == 2
+            assert_almost_equal(out[1,i,i], nd.array([0.02]), rtol=1e-3, atol=1e-5)
+        out.backward()
+        for i in range(LARGE_SQ_X):
```
Comment (access2rohit) [inline tests/nightly/test_large_array.py:1810]: Same

File: tests/nightly/test_large_array.py [hunk 3]
```diff
@@ -1791,6 +1792,29 @@ def test_sparse_dot():
...
+            # check the first row
+            assert A.grad[0,0,i] == 4
+            assert_almost_equal(A.grad[1,0,i], nd.array([0.4]), rtol=1e-3, atol=1e-5)
```
Comment (access2rohit) [inline tests/nightly/test_large_array.py:1813]: Question: Why did this become 0.4 and not 0.04 ? OR just let me know if this output os consistent with smaller inputs like 2x2 or 3x3.

**Comment by access2rohit** [comment]
Few comments ... overall code is good

File: tests/nightly/test_large_array.py (hunk 3 above)
Comment (Zha0q1) [reply tests/nightly/test_large_array.py:1813]: This is the correct result I believe. I verified with hand-written calculation. Yeah it also struck as counter-intuitive to me.. I am going to dive deep in matrix grad when I find time

Comment (access2rohit) [reply tests/nightly/test_large_array.py]: Can you just check with smaller input run and let me know the results. That should be good enough

**Comment by Zha0q1** [comment]
> Can you just check with smaller input run and let me know the results. That should be good enough .... overall LGTM!

This has been tested with both small and large input tensors

File: tests/nightly/test_large_array.py [hunk 4]
```diff
@@ -37,7 +37,7 @@
 LARGE_X = 100000000
 SMALL_X = 100
 SMALL_Y = 50
-LARGE_SQ_X = 80000
+LARGE_SQ_X = 70000
```
Comment (szha) [inline tests/nightly/test_large_array.py:40]: why?

Comment (ChaiBapchya) [reply tests/nightly/test_large_array.py:40]: 70000*70000/2**32 is just over 2**32
80k is lot more

Comment (Zha0q1) [reply tests/nightly/test_large_array.py:40]: We figured 7000 is large enough to overflow int 32. This is a new constant just introduced so the few of us tweak it to 70000

**Comment by ChaiBapchya** [comment]
@Zha0q1 also add your name to Contributors.md in the upcoming PR [let's not retrigger CI for that]. 

Thanks a lot for your contributions! :)

**Comment by Zha0q1** [comment]
> @Zha0q1 also add your name to Contributors.md in the upcoming PR [let's not retrigger CI for that].
> https://github.com/apache/incubator-mxnet/blob/master/CONTRIBUTORS.md
> Thanks a lot for your contributions! :)

will do!

File: tests/nightly/test_large_array.py (hunk 4 above)
Comment (access2rohit) [reply tests/nightly/test_large_array.py:40]: 70k * 70k = 4.9 billion 
int32 range < 4.3 billion
Therefore increasing input size will only increase test runtime.

**Pull request merged**